test/unittest/blas1/blas1_sdsdot_test.cpp

```diff
@@ -0,0 +1,131 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Dotright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a dot of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a dot of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_sdsdot_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+#include <iostream>
+#include <type_traits>
+
+enum class api_type : int { event = 0, result = 1 };
+
+/**
+ * Operator overload to print the test name properly
+ */
+std::ostream& operator<<(std::ostream& os, const api_type& type) {
+  if (type == api_type::event) {
+    return os << "event";
+  } else {
+    return os << "result";
+  }
+}
+
+template <typename scalar_t>
+using combination_t = std::tuple<api_type, int, float, int, int>;
+
+template <typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  /* sdsdot is only valid when using floats */
+  static_assert(std::is_same<scalar_t, float>::value);
+
+  index_t N;
+  float sb;
+  index_t incX;
+  index_t incY;
+  api_type api;
+  std::tie(api, N, sb, incX, incY) = combi;
+
+  using data_t = utils::data_storage_t<scalar_t>;
+
+  /* Sycl Buffers do not work with size = 0. So setting input vectors to size
+   * one to test the edge case where if size equals 0 then sb should be
+   * returned. */
+  index_t vectorSize = N > 0 ? N : 1;
```

Comment:
We claim that `N` can be zero in the docs. This then says that this doesn't work?

Reply:
Yes, N can be zero. However,  the size of the vectors cannot be zero. So,

vectorsSize > 0 and N == 0 // Works
vectorSize == 0 and N >= 0 // Does not work

The error occurs when creating the sycl buffer